include/library/spdm_common_lib.h

```diff
@@ -57,7 +57,8 @@ typedef enum {
     /* LIBSPDM_DATA_LOCAL_SLOT_COUNT is deprecated. Do not use it. Reserved only. */
     LIBSPDM_DATA_LOCAL_SLOT_COUNT,
     LIBSPDM_DATA_PEER_PUBLIC_ROOT_CERT,
-    LIBSPDM_DATA_PEER_PUBLIC_CERT_CHAIN,
+    LIBSPDM_DATA_LOCAL_PUBLIC_KEY,
```

Comment:
This issue only "add raw pub support". As such, it should not remove `LIBSPDM_DATA_PEER_PUBLIC_CERT_CHAIN`.

Reply:
My understanding from SPDM specification is that the certificate chain of peer should be and only be retrieved from `GET_CERTIFICATE` process, and the `LIBSPDM_DATA_PEER_PUBLIC_ROOT_CERT` should be the only one to be set base upon needs to help the verification of peer certificate chain.
Another thought is, it looks there is no place for the idea of "provisioned certificate chain of peer" because the leaf cert of the chain would be new-generated every time from the device secure-booting if it uses `alias_cert` mode - I guess that means a fixed, pre-set certificate chain of peer device may not be applicable.

Besides, currently the `LIBSPDM_DATA_PEER_PUBLIC_CERT_CHAIN` seems to be only for the slot id 0xFF (raw public key provisioning) related process. The only code that refers to `LIBSPDM_DATA_PEER_PUBLIC_CERT_CHAIN` I can find is in spdm-emu it sets `LIBSPDM_DATA_PEER_PUBLIC_CERT_CHAIN` for slot id 0xFF testing.
https://github.com/DMTF/spdm-emu/blob/5c85a16499ae4d4aa915b864727ba11572693291/spdm_emu/spdm_requester_emu/spdm_requester_spdm.c#L350-L369

This PR provides more appropriate type `LIBSPDM_DATA_LOCAL_PUBLIC_KEY` for raw public key process so I think `LIBSPDM_DATA_PEER_PUBLIC_CERT_CHAIN` can be removed.